src/gpu/generic/sycl/sycl_utils.hpp

```diff
@@ -35,6 +37,18 @@ inline bool md_dims_in_range(const dnnl::impl::memory_desc_t *desc) {
     return true;
 }
 
+inline ::sycl::nd_range<1> get_range(const exec_ctx_t &ctx, int work_amount) {
+    auto eng = dnnl::engine(ctx.stream()->engine(), true);
+    auto device = dnnl::sycl_interop::get_device(eng);
```

Comment:
Please don't use public API inside the library, use the internal API. 
```cpp
const auto *sycl_engine_impl = utils::downcast<const xpu::sycl::engine_impl_t *>(eng->impl());
auto device = sycl_engine_impl->device());
```

Reply:
Fixed